bigquery_storage/google/cloud/bigquery_storage_v1beta1/reader.py

```diff
@@ -43,6 +43,16 @@
 
 _STREAM_RESUMPTION_EXCEPTIONS = (google.api_core.exceptions.ServiceUnavailable,)
 
+# The Google API endpoint can unexpectedly close long-running HTTP/2 streams.
+# Unfortunately, this condition is surfaced to the caller as an internal error
+# by gRPC. We don't want to resume on all internal errors, so instead we look
+# for error message that we know are caused by problems that are safe to
+# reconnect.
+_STREAM_RESUMPTION_INTERNAL_ERROR_MESSAGES = (
+    # See: https://issuetracker.google.com/143292803
+    "unexpected EOS on DATA frame",
```

Comment:
I don't know that the error messages surfaced by gRPC on unexpected HTTP/2 stream termination are the same here from one language to another. Can we confirm this?

Reply:
There isn't any message rewriting going on in the Python client.

https://github.com/googleapis/google-cloud-python/blob/master/api_core/google/api_core/exceptions.py#L447-L462

Java appears to be resurfacing the `io.grpc.StatusRuntimeException` gRPC exception directly, so it should have the same message.

Reply:
It appears that this error is generated client-side: https://github.com/grpc/grpc-java/blob/81da3eb95be37fa0647ce8da2e19de96ab84c600/core/src/main/java/io/grpc/internal/Http2ClientStreamTransportState.java#L151

So, you're right that we need to check what the equivalent error is in Python.

Reply:
This should be updated based on the new test result, right?

Reply:
Done in 50ecc98466c1487561124bd149ac997c0de566e9

Reply:
How stable is the server-side error message? It is _probably_ not expected to change often (if at all)?

Reply:
According to the blame view, the line that generates this message was last updated 3 years ago, and that was just to format the code.

https://github.com/grpc/grpc/blame/9dd05c4abc1741223ded8566bc143c927d9cba79/src/core/ext/transport/chttp2/transport/frame_rst_stream.cc#L110

Reply:
Sounds good.